edgelet/edgelet-http/src/util/hyperwrap.rs

```diff
@@ -36,17 +51,39 @@ impl Config {
         if self.null {
             Ok(Client::Null)
         } else {
-            let config = self.clone();
-
-            let https =
-                HttpsConnector::new(DNS_WORKER_THREADS).context(ErrorKind::Initialization)?;
+            let https_connector = match &self.identity_certificate {
+                Some(id) => {
+                    let identity = id.get_identity()?;
+                    let mut builder = TlsConnector::builder();
+                    if let Some(bundle) = &self.trust_bundle {
+                        let certs = X509::stack_from_pem(&bundle.cert)
+                            .with_context(|_| ErrorKind::TrustBundle)?;
+                        for cert in certs {
+                            let der = cert.to_der().with_context(|_| ErrorKind::TrustBundle)?;
+                            let c = TlsCertificate::from_der(&der)
+                                .with_context(|_| ErrorKind::TrustBundle)?;
+                            builder.add_root_certificate(c);
+                        }
+                    }
+                    let connector = builder
+                        .identity(identity)
+                        .build()
+                        .with_context(|_| ErrorKind::CertificateConversionError)?;
+                    let mut http = HttpConnector::new(DNS_WORKER_THREADS);
+                    http.enforce_http(false);
+                    Ok(HttpsConnector::from((http, connector)))
+                }
+                None => HttpsConnector::new(DNS_WORKER_THREADS).context(ErrorKind::Initialization),
```

Comment:
Don't you want to check for `self.trust_bundle` here too?

Reply:
This is handled in lines 58-67 which handles the case if a trust bundle were to be provided.

Reply:
Lines 58-67 handle it when an identity cert is also set. I'm asking about the case where an identity cert isn't set.

Reply:
Ah yes, that is missing. I will add it.